### PR TITLE
Issue #20 - Network Settings: fix allow customizable minimum player count for game start

### PR DESCRIPTION
**Pull request by jovanHillion**
### Description
This fix add a scriptable object able to configure some rooms settings, also add a network utils static class to avoid duplicate code, few changes in simple test to use utils.
Also add some verification for the scriptable object configuration
Able Players to start a game when not full in a room depending on the settings

### Related Issue(s)
Issue #20 

### Changes Made
- Add a scriptable object for network settings
- Adapt 2 unit tests
- Add some check depending on settings with the network
- Able player to start a game when not full in a room

### Testing
- Functional tests have been done
- Unit tests passed successfully

### Screenshots (if applicable)
Capture d'écran 2024-10-17 114542
Capture d'écran 2024-10-17 114605
Capture d'écran 2024-10-17 114612
Capture d'écran 2024-10-17 114639
Capture d'écran 2024-10-17 114705
Capture d'écran 2024-10-17 114746
Capture d'écran 2024-10-17 114752

### Checklist
- [x] I have tested these changes thoroughly.
- [x] The code follows the project's style guide and coding conventions.
- [x] All tests passed successfully.
- [x] I have checked for any potential conflicts with other branches.

### Reviewer(s) (optional)
@PharaEthan

### Definition of Done
Network is configurable using Scriptable Object
We can join a room even if it is not full
Unit tests passed successfully

### CODE REVIEW

File: Vermines/Assets/Scripts/Network/Settings/NetworkSettings.cs [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+    /*
+     * @brief keyStringCodeGeneration is used to store the key string to generate random codes.
+     */
+    public string keyStringCodeGeneration = "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
+
+    /*
+     * @brief random is used to generate random numbers if needed.
+     */
+    public System.Random random = new();
```
Comment (PharaEthan) [inline Vermines/Assets/Scripts/Network/Settings/NetworkSettings.cs:58]: Il y a vraiment besoin de ça ?

File: Vermines/Assets/Scripts/Network/Settings/NetworkSettings.cs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+    /*
+     * @brief enableDebugLogging is used to enable or disable debug logging.
+     */
+    public bool enableDebugLogging = true;
```
Comment (PharaEthan) [inline Vermines/Assets/Scripts/Network/Settings/NetworkSettings.cs:63]: Pareil pour ça ?

**Review by PharaEthan — APPROVED** [review]

File: Vermines/Assets/Scripts/Network/Settings/NetworkSettings.cs (hunk 2 above)
Comment (jovanHillion) [reply Vermines/Assets/Scripts/Network/Settings/NetworkSettings.cs:63]: à l'avenir si on à beaucoup de log ca me semble être utile, a moins qu'il y ait une meilleure option

File: Vermines/Assets/Scripts/Network/Settings/NetworkSettings.cs (hunk 1 above)
Comment (jovanHillion) [reply Vermines/Assets/Scripts/Network/Settings/NetworkSettings.cs:58]: En effet, après refexion elle ne devrait pas être la, je la bougerai à l'avenir

File: Vermines/Assets/Scripts/Network/Settings/NetworkSettings.cs (hunk 2 above)
Comment (PharaEthan) [reply Vermines/Assets/Scripts/Network/Settings/NetworkSettings.cs:63]: Je disais ça parce que les Debug.Log ne marche seulement qu'en débug, ducoup je voyais pas trop l'intérêt

**Pull request merged**